tests/lib/rules/multiline-comment-style.js

```diff
@@ -482,6 +712,348 @@ ruleTester.run("multiline-comment-style", rule, {
+                   bar */
+            `,
+            options: ["bare-block"],
+            errors: [{ messageId: "expectedBareBlock", line: 2 }]
+        },
+        {
+            code: `
+                /*
+                 *    foo
+                 *  bar
+                 *   baz
+                 * qux
+                 */
+            `,
+            output: `
+                /*    foo
+                    bar
+                     baz
+                   qux */
+            `,
+            options: ["bare-block"],
+            errors: [{ messageId: "expectedBareBlock", line: 2 }]
+        },
+        {
+            code: `
+                /*
+                {
+                    "foo": 1,
+                    "bar": 2
+                }
+                */
+            `,
+            output: `
+                /*
+                 * {
+                 *     "foo": 1,
+                 *     "bar": 2
+                 * }
+                 */
+            `,
+            errors: [
+                { messageId: "missingStar", line: 3 },
+                { messageId: "missingStar", line: 4 },
+                { messageId: "missingStar", line: 5 },
+                { messageId: "missingStar", line: 6 },
+                { messageId: "alignment", line: 7 }
+            ]
+        },
+        {
+            code: `
+                /*
+                {
+                \t"foo": 1,
+                \t"bar": 2
+                }
+                */
+            `,
+            output: `
+                /*
+                 * {
+                 * \t"foo": 1,
+                 * \t"bar": 2
+                 * }
+                 */
+            `,
+            errors: [
+                { messageId: "missingStar", line: 3 },
+                { messageId: "missingStar", line: 4 },
+                { messageId: "missingStar", line: 5 },
+                { messageId: "missingStar", line: 6 },
+                { messageId: "alignment", line: 7 }
+            ]
+        },
+        {
+            code: `
+                /*
+                {
+                \t  "foo": 1,
+                \t  "bar": 2
+                }
+                */
+            `,
+            output: `
+                /*
+                 * {
+                 * \t  "foo": 1,
+                 * \t  "bar": 2
+                 * }
+                 */
+            `,
+            errors: [
+                { messageId: "missingStar", line: 3 },
+                { messageId: "missingStar", line: 4 },
+                { messageId: "missingStar", line: 5 },
+                { messageId: "missingStar", line: 6 },
+                { messageId: "alignment", line: 7 }
+            ]
+        },
+        {
+            code: `
+                /*
+                {
+               \t"foo": 1,
+               \t"bar": 2
+                }
+                */
+            `,
+            output: `
+                /*
+                 * {
+                 * "foo": 1,
```

Comment:
Should these lines also have tabs? Or are they ignored because the tabs in the input are aligned with the `/` rather than the `*` on the first line of the comment?

Reply:
The current autofixer moves any lines that have less indentation than the opening comment delimiter up to minimum indentation required to align with the `*` characters on each line (in other words, removing all leading whitespace and adding a space after the `*`). I don't know how we can can compare lines that have mixed tabs and spaces in their leading whitespace, so I just used the number of whitespace characters to compare. We could alternatively use all the whitespace from the beginning of the line to the first non-whitespace character, but I think the current behavior is more in line with what folks would expect. Definitely open to other options if someone has a better suggestion of how to handle this!